src/guide/data_types.md

```diff
@@ -183,3 +183,36 @@ subtractOne(counter)
 
 print(counter) // { count: 1 }
 ```
+
+## Mutually Recursive Data Types
+
+In some instances, it is necessary to define multiple data types which refer to one another. For example, imagine you wanted to define your own number system based on [Peano Arithmetic](https://en.wikipedia.org/wiki/Peano_axioms). You might try something like this:
```

Comment:
I dunno why, but reading "one another" made me think that I couldn't do something like defining a type and then another type that refers to the first one. I think this makes it more explicit what we're talking about here:
```suggestion
In some instances, it's necessary to define multiple data types where each of the types refers to the others. For example, imagine you wanted to define your own number system based on [Peano Arithmetic](https://en.wikipedia.org/wiki/Peano_axioms). You might try something like this:
```

Reply:
Maybe `which refer to each other` would be more clear?

Reply:
I think that would be fine 👍